lib/fizzy/execute.cpp

```diff
@@ -138,16 +138,21 @@ inline bool store_into_memory(
 inline uint32_t grow_memory(
     bytes& memory, uint32_t delta_pages, uint32_t memory_pages_limit) noexcept
 {
+    // Guaranteed by instantiate allocation + resize in grow_memory.
+    assert(memory.size() % PageSize == 0);
```

Comment:
BTW for imported memory this is not guaranteed. Niether that its size is inside the limits (only its declared size is checked).
I guess instantiate should check that, too.

Reply:
Hmm, do you have a reference in the spec which allows imports to be not pagesize-aligned?

Reply:
By "not guaranteed" I meant that our instantiate currently doesn't enforce this, but I think it should.

In the spec it is required in the valid external memory definition https://webassembly.github.io/spec/core/exec/modules.html#xref-exec-runtime-syntax-externval-mathsf-mem-a:

![image](https://user-images.githubusercontent.com/1863135/109699977-83676d80-7b91-11eb-8d4b-d3eabe821db1.png)

and in memory definition https://webassembly.github.io/spec/core/exec/runtime.html#memory-instances:

![image](https://user-images.githubusercontent.com/1863135/109700171-bb6eb080-7b91-11eb-800b-5cd70022d8ec.png)

Reply:
It is unfortunate that the spectest architecture can not describe such test cases (i.e. imported memory not a multiple of pagesize)

Reply:
> BTW for imported memory this is not guaranteed. Niether that its size is inside the limits (only its declared size is checked).

I've added the alignment check in #749 
The check for size being inside limits was already done apparently.